pkg/ntpdate/ntpdate.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2021 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package ntpdate
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/beevik/ntp"
+
+	"github.com/u-root/u-root/pkg/rtc"
+)
+
+type debugFn func(string, ...interface{})
+
+func parseServers(r *bufio.Reader, debug debugFn) []string {
+	var uri []string
+	var l string
+	var err error
+
+	debug("Reading config file")
+	for err == nil {
+		// This handles the case where the last line doesn't end in \n
+		l, err = r.ReadString('\n')
+		debug("%v", l)
+		if w := strings.Fields(l); len(w) > 1 && w[0] == "server" {
+			// We look only for the server lines, we ignore options like iburst
+			// TODO(ganshun): figure out what options we want to support.
+			uri = append(uri, w[1])
+		}
+	}
+
+	return uri
+}
+
+func getTime(servers []string, debug debugFn) (time.Time, string, error) {
+	for _, s := range servers {
+		debug("Getting time from %v", s)
+		t, err := ntp.Time(s)
+		if err == nil {
+			// Right now we return on the first valid time.
+			// We can implement better heuristics here.
+			debug("Got time %v", t)
+			return t, s, nil
+		}
+		debug("Error getting time from %s: %v", s, err)
+	}
+
+	return time.Time{}, "", fmt.Errorf("unable to get any time from servers %v", servers)
+}
+
+// SetTime sets system and optionally RTC time from NTP servers specified in sersers or the config file.
+func SetTime(servers []string, config string, fallback string, setRTC bool, verbose bool) error {
+	servers = servers[:]
+
+	debug := func(string, ...interface{}) {}
+	if verbose {
+		debug = log.Printf
```

Comment:
A more common convention in u-root is to make a package global, Debug:
var Debug = func(string,...interface{}} {}
which can be set outside the package in, e.g., main(). This would allow you to have one less parameter on SetTime, but more important allows users of the package to control where Debug output appears. This is particularly handy in tests:
Debug = t.Logf
but, further, avoids having a library spew debug code on os.Stderr via log.Printf.

I hope this change would be ok for you.